src/ethereum/EnigmaContractProductionWriterAPI.js

```diff
@@ -308,19 +308,21 @@ class EnigmaContractProductionWriterAPI extends EnigmaContractWriterAPI {
       const blockNumber = await utils.getEthereumBlockNumber(this.w3());
 
       const resolveLogic = async () => {
-        let rawEvents = await this._enigmaContract.getPastEvents("allEvents", {
-          fromBlock: blockNumber,
-          filter: { taskId: utils.add0x(taskId) }
-        });
-        if (Array.isArray(rawEvents) && rawEvents.length > 0) {
-          rawEvents.forEach(event => {
-            if (
-              event.event === constants.RAW_ETHEREUM_EVENTS.ReceiptFailedETH ||
-              event.event === constants.RAW_ETHEREUM_EVENTS.ReceiptVerified
-            ) {
-              resolve(this._parseEvents({ [event.event]: event }));
-            }
-          });
+        const filter = { taskId: utils.add0x(taskId) };
```

Comment:
Why not call `allEvents` once and then check the events name (true both for this function and the `deploySecretContract`)?

Reply:
This will also work. But it'll return an `EventEmitter` instead of an array... I'll try to work this into `_parsePastEvents`.

Reply:
Also, we will have to filter ourselves by `eventName` as `allEvents` returns all the events associated with the contract and filtering by `taskId` may not be enough.

So I'm not sure it worth changing the code for this.

Reply:
My bad, you meant using `_parsePastEvents("allEvents")`. I thought you meant using `this._enigmaContract.allEvents()` which returns an `EventEmitter`. Sorry, will fix. :pray: